compiler/lib/global_flow.mli

```diff
@@ -31,10 +31,17 @@ type approx =
       ; others : bool (* Whether other values are possible *)
       }
 
+type escape_status =
+  | Escape
+  | Escape_constant (* Escapes but we know the value is not modified *)
+  | No
+
 type info =
   { info_defs : def array
   ; info_approximation : approx Var.Tbl.t
   ; info_may_escape : Var.ISet.t
```

Comment:
Do we need to keep info_may_escape ? Or can't we just change its type ?

Reply:
@vouillon

Reply:
We need both.
`info_may_escape` are a set of _values_ that may escape. It is used for the CPS transformation (functions that escape must be in CPS).
`info_variable_may_escape` is a set of _variables_ whose contents may escape. For dead code elimination, we use it to tell whether the arguments of a function may escape.

Reply:
Ok, I mis-read that `info_may_escape` was derived from `variable_may_escape` but it's in fact derived from `may_escape`.